spec/datashake-ruby-sdk/review_scraper/client_spec.rb

```diff
@@ -3,7 +3,25 @@
 require "spec_helper"
 
 RSpec.describe Datashake::ReviewScraper::Client do
-  subject { described_class.new(token: "abc") }
+  subject { described_class.new(token: "abc", endpoint: endpoint) }
+
+  let(:endpoint) { nil }
+
+  describe "#initialize" do
+    context "when endpoint is provided" do
+      let(:endpoint) { "https://custom.endpoint.com" }
+
+      it "uses the provided endpoint as the base_url" do
+        expect(subject.instance_variable_get(:@base_url)).to eq(endpoint)
```

Comment:
Thanks for adding specs! I'm wondering if we can test not just an instance variable here, but the actual url_prefix we use for the connection? something like `client.connection.url_prefix` maybe? WDYT?

Reply:
Thanks good idea, I changed it 🙂